Appender does not appear to support null timestamps

Here is a test that shows the failure:

```go
func TestAppenderNullTimestampTZ(t *testing.T) {
	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (ts TIMESTAMPTZ)`)
	defer cleanupAppender(t, c, db, conn, a)

	// Append a nil *time.Time.
	var nilTime *time.Time
	require.NoError(t, a.AppendRow(nilTime))
	require.NoError(t, a.Flush())

	// Verify results.
	res := db.QueryRowContext(context.Background(), `SELECT ts FROM test`)

	var r *time.Time
	require.NoError(t, res.Scan(&r))
	require.Nil(t, r)
}
```

Relevant output:

```
go test -run=TestAppenderNullTimestampTZ
--- FAIL: TestAppenderNullTimestampTZ (0.02s)
                Error:          Received unexpected error:
                                database/sql/driver: could not append row: failed to set value: at row 0, col 0, val: <nil>: cast error: cannot cast *time.Time to time.Time
                Test:           TestAppenderNullTimestampTZ
FAIL
exit status 1
FAIL    github.com/duckdb/duckdb-go/v2  0.465s
```

It appears that when the column type is determined and the mapping is established, it does not account for a timestamp to be nullable and thus assumes a `time.Time` value versus a `*time.Time` type.

Note: this may applies to the `time` type as well and I only tried this with the appender API, not through standard inserts.

--- a/appender_test.go
+++ b/appender_test.go
@@ -580,6 +580,108 @@ func TestAppenderTime(t *testing.T) {
 	require.Equal(t, base.UnixMicro(), r.UnixMicro())
 }
 
+func TestAppenderNullTimestampTZ(t *testing.T) {
+	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (ts TIMESTAMPTZ)`)
+	defer cleanupAppender(t, c, db, conn, a)
+
+	// Append a nil *time.Time.
+	var nilTime *time.Time
+	require.NoError(t, a.AppendRow(nilTime))
+	require.NoError(t, a.Flush())
+
+	// Verify results.
+	res := db.QueryRowContext(context.Background(), `SELECT ts FROM test`)
+
+	var r *time.Time
+	require.NoError(t, res.Scan(&r))
+	require.Nil(t, r)
+}
+
+func TestAppenderNullDate(t *testing.T) {
+	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (d DATE)`)
+	defer cleanupAppender(t, c, db, conn, a)
+
+	// Append a nil *time.Time.
+	var nilTime *time.Time
+	require.NoError(t, a.AppendRow(nilTime))
+	require.NoError(t, a.Flush())
+
+	// Verify results.
+	res := db.QueryRowContext(context.Background(), `SELECT d FROM test`)
+
+	var r *time.Time
+	require.NoError(t, res.Scan(&r))
+	require.Nil(t, r)
+}
+
+func TestAppenderNullTime(t *testing.T) {
+	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (t TIME)`)
+	defer cleanupAppender(t, c, db, conn, a)
+
+	// Append a nil *time.Time.
+	var nilTime *time.Time
+	require.NoError(t, a.AppendRow(nilTime))
+	require.NoError(t, a.Flush())
+
+	// Verify results.
+	res := db.QueryRowContext(context.Background(), `SELECT t FROM test`)
+
+	var r *time.Time
+	require.NoError(t, res.Scan(&r))
+	require.Nil(t, r)
+}
+
+func TestAppenderNullTimeTZ(t *testing.T) {
+	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (t TIMETZ)`)
+	defer cleanupAppender(t, c, db, conn, a)
+
+	// Append a nil *time.Time.
+	var nilTime *time.Time
+	require.NoError(t, a.AppendRow(nilTime))
+	require.NoError(t, a.Flush())
+
+	// Verify results.
+	res := db.QueryRowContext(context.Background(), `SELECT t FROM test`)
+
+	var r *time.Time
+	require.NoError(t, res.Scan(&r))
+	require.Nil(t, r)
+}
+
+func TestAppenderNullInterval(t *testing.T) {
+	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (i INTERVAL)`)
+	defer cleanupAppender(t, c, db, conn, a)
+
+	// Append a nil *Interval.
+	var nilInterval *Interval
+	require.NoError(t, a.AppendRow(nilInterval))
+	require.NoError(t, a.Flush())
+
+	// Verify results.
+	res := db.QueryRowContext(context.Background(), `SELECT i FROM test`)
+
+	var r *Interval
+	require.NoError(t, res.Scan(&r))
+	require.Nil(t, r)
+}
+
+func TestAppenderNullHugeInt(t *testing.T) {
+	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (h HUGEINT)`)
+	defer cleanupAppender(t, c, db, conn, a)
+
+	// Append a nil *big.Int.
+	var nilBigInt *big.Int
+	require.NoError(t, a.AppendRow(nilBigInt))
+	require.NoError(t, a.Flush())
+
+	// Verify results.
+	res := db.QueryRowContext(context.Background(), `SELECT h FROM test`)
+
+	var r *big.Int
+	require.NoError(t, res.Scan(&r))
+	require.Nil(t, r)
+}
+
 func TestAppenderTimeTZ(t *testing.T) {
 	c, db, conn, a := prepareAppender(t, `CREATE TABLE test (time TIMETZ)`)
 	defer cleanupAppender(t, c, db, conn, a)

--- a/vector.go
+++ b/vector.go
@@ -1,6 +1,8 @@
 package duckdb
 
 import (
+	"math/big"
+	"time"
 	"unsafe"
 
 	"github.com/duckdb/duckdb-go/mapping"
@@ -179,7 +181,7 @@ func (vec *vector) initTS(t Type) {
 		return vec.getTS(t, rowIdx)
 	}
 	vec.setFn = func(vec *vector, rowIdx mapping.IdxT, val any) error {
-		if val == nil {
+		if val == nil || val == (*time.Time)(nil) {
 			vec.setNull(rowIdx)
 			return nil
 		}
@@ -196,7 +198,7 @@ func (vec *vector) initDate() {
 		return vec.getDate(rowIdx)
 	}
 	vec.setFn = func(vec *vector, rowIdx mapping.IdxT, val any) error {
-		if val == nil {
+		if val == nil || val == (*time.Time)(nil) {
 			vec.setNull(rowIdx)
 			return nil
 		}
@@ -213,7 +215,7 @@ func (vec *vector) initTime(t Type) {
 		return vec.getTime(rowIdx)
 	}
 	vec.setFn = func(vec *vector, rowIdx mapping.IdxT, val any) error {
-		if val == nil {
+		if val == nil || val == (*time.Time)(nil) {
 			vec.setNull(rowIdx)
 			return nil
 		}
@@ -230,7 +232,7 @@ func (vec *vector) initInterval() {
 		return vec.getInterval(rowIdx)
 	}
 	vec.setFn = func(vec *vector, rowIdx mapping.IdxT, val any) error {
-		if val == nil {
+		if val == nil || val == (*Interval)(nil) {
 			vec.setNull(rowIdx)
 			return nil
 		}
@@ -247,7 +249,7 @@ func (vec *vector) initHugeint() {
 		return vec.getHugeint(rowIdx)
 	}
 	vec.setFn = func(vec *vector, rowIdx mapping.IdxT, val any) error {
-		if val == nil {
+		if val == nil || val == (*big.Int)(nil) {
 			vec.setNull(rowIdx)
 			return nil
 		}

